models/m_audit.erl

```diff
@@ -131,6 +135,44 @@ user_agent_id(UserAgent, Context) ->
     Transaction = fun() -> z_db:transaction(F, Context) end,
     z_depcache:memo(Transaction, {user_agent_id, BUA}, ?MAXAGE_UA_STRING, Context).
 
+% Remove entries older than 2 years.
+periodic_cleanup(Context) ->
+    z_db:q("delete from audit where id in (select id from audit where created < now() - interval '2 years' limit 10000)", Context, 300000).
+
+
+% Update all records in the audit table
+update_audit_table(Context) ->
+    io:fwrite(standard_error, "Updating audit table", []),
+    update_audit_table1(Context).
+
+update_audit_table1(Context) ->
+    case props_json_update(Context) of
+        {ok, N} when N > 0 ->
+            io:fwrite(standard_error, ".", []),
+            update_audit_table1(Context);
+        {ok, 0} ->
+            io:fwrite(standard_error, ".~n", []),
+            done
+    end.
+
+% Move the stored props of erlang term props to json term props.
+props_json_update(Context) ->
+    F = fun(Ctx) ->
+                L = [ok = audit_record_to_prop_json(Id, Ctx) || {Id} <- z_db:q("select id from audit where props is not null and props_json is null limit 100", Ctx)],
```

Comment:
Maybe this `select` could be moved to it's own, named function to provide better context? Something like `get_non_json_props`, or `get_untransferred_props`?

Reply:
That doesn't make it simple enough to defend the addition of another function I think. I might probably do that when the function has to become a generic and has to be able to upgrade any table.